integration/test/test_cpu_activity_agent.py

```diff
@@ -0,0 +1,269 @@
+            if key not in ("CPU_FREQ_MAX", "CPU_FREQ_EFFICIENT", "CPU_UNCORE_FREQ_MAX", "CPU_UNCORE_FREQ_EFFICIENT", "CPU_PHI"):
+                uncore_mbm_list.append(value)
+
+        #######################################################
+        # Core frequency sweep at fixed uncore_efficient_freq #
+        #######################################################
+        cls._aib_core_freq_sweep_dir = Path(os.path.join('test_cpu_activity_output', 'core_frequency_sweep'))
+        experiment_args.output_dir = cls._aib_core_freq_sweep_dir
+        experiment_args.min_frequency = cpu_min_freq
+        experiment_args.max_frequency = cpu_max_freq
+        experiment_args.min_uncore_frequency = uncore_efficient_freq
+        experiment_args.max_uncore_frequency = uncore_efficient_freq
+
+        launch_helper(uncore_frequency_sweep, experiment_args, aib_app_conf, experiment_cli_args)
+
+        ##############
+        # Parse data #
+        ##############
+        df_frequency_sweep = geopmpy.io.RawReportCollection('*report', dir_name=cls._aib_core_freq_sweep_dir).get_df()
+        policy = gen_cpu_activity_policy_recommendation.main(df_frequency_sweep, ['intensity_1', 'intensity_16'])
+        cpu_efficient_freq = policy['CPU_FREQ_EFFICIENT']
+
+        ################################
+        # CPU Activity Agent phi sweep #
+        ################################
+        geopm_test_launcher.geopmwrite("CPU_FREQUENCY_MAX_CONTROL board 0 {}".format(cpu_max_freq))
+        geopm_test_launcher.geopmwrite("CPU_UNCORE_FREQUENCY_MAX_CONTROL board 0 {}".format(uncore_max_freq))
+        geopm_test_launcher.geopmwrite("CPU_UNCORE_FREQUENCY_MIN_CONTROL board 0 {}".format(uncore_min_freq))
+
+        # Arithmetic Intensity Benchmark
+        cls._aib_agent_dir = Path(os.path.join('test_cpu_activity_output', 'aib_cpu_activity'))
+
+        ca_experiment_args = SimpleNamespace(
+            output_dir=cls._aib_agent_dir,
+            node_count=node_count,
+            trial_count=3,
+            cool_off_time=3,
+            enable_traces=False,
+            enable_profile_traces=False,
+            verbose=False,
+            cpu_fe=cpu_efficient_freq,
+            cpu_fmax=cpu_max_freq,
+            uncore_fe=uncore_efficient_freq,
+            uncore_fmax=cpu_max_freq,
+            uncore_mbm_list=uncore_mbm_list,
+            phi_list=[0.2, 0.5, 0.7],
+        )
+
+        launch_helper(cpu_activity, ca_experiment_args, aib_app_conf, [])
+
+        # MiniFE
+        cls._minife_agent_dir = Path(os.path.join('test_cpu_activity_output', 'minife_cpu_activity'))
+        ca_experiment_args.output_dir = cls._minife_agent_dir
+        ca_experiment_args.trial_count = 1
+
+        launch_helper(cpu_activity, ca_experiment_args, minife.MinifeAppConf(node_count), [])
+
+    def tearDown(self):
+        if sys.exc_info() != (None, None, None):
+            TestIntegration_cpu_activity._keep_files = True
+
+    @util.skip_unless_config_enable('beta')
```

Comment:
This should be added up at the class level (and removed from here) to ensure that setUpClass() does not run if beta is not enabled.  Seems like it would take a long time given how many combinations of agents and runs there are.